### PR TITLE
Feature/wtp 23 cmake

### PR DESCRIPTION
**Pull request by matthias-stone**
Changes the build system to CMake to simplify adding cross-platform library installation.

### CODE REVIEW

**Comment by stellentus** [comment]
I've tested this (following the updated README) and the tests pass on my machine.

File: test/05-tag-locking.c [hunk 1]
```diff
@@ -41,7 +39,7 @@ main(int argc, char** argv)
     //plc_tag_set_debug_level(PLCTAG_DEBUG_SPEW);
 
     for (i = 0; i < sizeof(threads) / sizeof(threads[0]); i++) {
-        if (pthread_create(&threads[i], NULL, thread_entry, NULL)) {
+        if (pthread_create(&threads[i], NULL, thread_entry, (uintptr_t)i)) {
```
Comment (dijkstracula) [inline test/05-tag-locking.c]: This is warning on clang:

```
05-tag-locking.c:42:61: warning: incompatible integer to pointer conversion passing 'uintptr_t' (aka 'unsigned long') to
      parameter of type 'void *' [-Wint-conversion]
        if (pthread_create(&threads[i], NULL, thread_entry, (uintptr_t)i)) {
```

**Review by dijkstracula — COMMENTED** [review]
Looks great, modulo the one warning.  Shall we also remove the old Makefile, too?

File: test/05-tag-locking.c (hunk 1 above)
Comment (matthias-stone) [reply test/05-tag-locking.c]: Fixed, thanks.

Comment (matthias-stone) [reply test/05-tag-locking.c]: Also removed the Makefiles - typed too fast and didn't realize `git --rm` had failed. Apparently I can't see stderr today...

**Pull request merged**